examples/architectureComponentsExample/src/main/java/io/realm/examples/arch/ArchExampleActivity.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.arch;
+
+import android.os.Bundle;
+import android.support.annotation.MainThread;
+import android.support.design.widget.FloatingActionButton;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+
+
+public class ArchExampleActivity extends AppCompatActivity {
+    private FloatingActionButton backgroundJobStartStop;
+
+    private BackgroundTask backgroundTask;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_arch_example);
+        setupViews();
+
+        backgroundTask = (BackgroundTask) getLastCustomNonConfigurationInstance();
+        if (backgroundTask == null) { // this could also live inside a ViewModel, a singleton job queue, etc.
```

Comment:
I managed to get into a state where two background updater threads were running at the same time. I'm not 100% how it happened. 

But doing:

1) Click details for Christian
2) Press back
3) Press back (exit app)
4) Click app (opens it again)
5) Increments now happen at double speed.

The thread isn't stopped in `onBackPressed` so my guess is that it keeps running somehow.

Reply:
Gah, I didn't notice this! Thanks!